src/core/deployer.sol

```diff
@@ -12,187 +11,68 @@
-        principal = new Principal();
-        principal.rely(msg.sender);
-        principal.deny(address(this));
-    }
-}
+        // Borrower  Depends
+        borrowerDeployer.collector().depend("distributor", distributor_);
+        // todo needs to be tranche after mock
+        borrowerDeployer.shelf().depend("lender", distributor_);
+        borrowerDeployer.collector().depend("distributor", distributor_);
 
-contract ThresholdFab {
-    function newThreshold() public returns (PushRegistry threshold) {
-        threshold = new PushRegistry();
-        threshold.rely(msg.sender);
-        threshold.deny(address(this));
+        //  Lender  depends
+        lenderDeployer.distributor().depend("shelf", shelf_);
     }
-}
-
-contract Deployer {
-    TitleFab          titlefab;
-    LightSwitchFab    lightswitchfab;
-    ShelfFab          shelffab;
-    DistributorFab    distributorFab;
-    PileFab           pilefab;
-    PrincipalFab      principalFab;
-    CollectorFab      collectorFab;
-    ThresholdFab      thresholdFab;
-
-    address     public god;
-
-    Title       public title;
-    LightSwitch public lightswitch;
-    Shelf       public shelf;
-    Distributor  public distributor;
-    LenderLike  public lender;
-    Pile        public pile;
-    Principal   public principal;
-    Collector   public collector;
-    PushRegistry public threshold;
 
+    function wireDeployment() public auth {
+        require(address(borrowerDeployer) != address(0));
+        require(address(lenderDeployer) != address(0));
 
-    constructor (address god_, TitleFab titlefab_, LightSwitchFab lightswitchfab_, ShelfFab shelffab_, DistributorFab distributorFab_, PileFab pilefab_, PrincipalFab principalFab_, CollectorFab collectorFab_, ThresholdFab thresholdFab_) public {
-        god = god_;
 
-        titlefab = titlefab_;
-        lightswitchfab = lightswitchfab_;
-        shelffab = shelffab_;
-        distributorFab = distributorFab_;
-        pilefab = pilefab_;
-        principalFab = principalFab_;
-        collectorFab = collectorFab_;
-        thresholdFab = thresholdFab_;
-    }
-
-    function deployThreshold() public {
-        threshold = thresholdFab.newThreshold();
-        threshold.rely(god);
-
-    }
-    function deployCollector() public {
-        collector = collectorFab.newCollector(address(distributor), address(shelf), address(pile), address(threshold));
-        collector.rely(god);
-    }
-
-    function deployPile() public {
-        pile = pilefab.newPile();
-        pile.rely(god);
-    }
-
-    function deployTitle(string memory name, string memory symbol) public {
-        title = titlefab.newTitle(name, symbol);
-        title.rely(god);
-    }
-
-    function deployLightSwitch() public {
-        lightswitch = lightswitchfab.newLightSwitch();
-        lightswitch.rely(god);
-    }
+        // distributor allowed to call
+        borrowerDeployer.shelf().rely(address(lenderDeployer.distributor()));
```

Comment:
don't think that this is required. distributor just calls public methods from shelf